vertx-sql-client-template/src/main/java/io/vertx/sqlclient/template/generator/RowMapperGen.java

```diff
@@ -0,0 +1,215 @@
+            }
+            break;
+          }
+        }
+      });
+    model
+      .getPropertyMap()
+      .values()
+      .stream()
+      .filter(prop -> PK.contains(prop.getKind()))
+      .filter(prop -> prop.isAdder() && !prop.isSetter())
+      .forEach(prop -> {
+        String meth = getArrayType(prop.getType());
+        String rowType = rowType(prop.getType());
+        if (meth != null) {
+          String columnName = getMappingName(prop, Column.class.getName());
+          if (columnName != null) {
+            writer.print("    val = row." + meth + "(\"" + columnName + "\");\n");
+            writer.print("    if (val != null) {\n");
+            writer.print("      for (" + rowType + " elt : (" + rowType + "[])val) {\n");
+            writer.print("        obj." + prop.getAdderMethod() + "(" + wrapExpr(prop.getType(), "elt") + ");\n");
+            writer.print("      }\n");
+            writer.print("    }\n");
+          }
+        }
+      });
+  }
+
+  private void bilto4(PrintWriter writer, String meth, PropertyInfo prop, String converter) {
+    String columnName = getMappingName(prop, Column.class.getName());
+    if (columnName != null) {
+      writer.print("    val = row." + meth + "(\"" + columnName + "\");\n");
+      writer.print("    if (val != null) {\n");
+      writer.print("      obj." + prop.getSetterMethod() + "(" + converter +  ");\n");
+      writer.print("    }\n");
+    }
+  }
+
+  private static String wrapExpr(TypeInfo type, String expr) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      MapperInfo deserializer = dataObject.getDeserializer();
+      if (deserializer != null) {
+        if (deserializer.getKind() == MapperKind.SELF) {
+          return "new " + type.getName() + "(" + expr + ")";
+        } else {
+          return deserializer.getQualifiedName() + "." + String.join(".", deserializer
+            .getSelectors()) + "(" + expr + ")";
+        }
+      }
+      throw new UnsupportedOperationException();
+    } else {
+      return expr;
+    }
+  }
+
+  private static String rowType(TypeInfo type) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      return rowType(dataObject.getJsonType());
+    }
+    return type.getName();
+  }
+
+  private static String getter(TypeInfo type) {
+    switch (type.getKind()) {
+      case PRIMITIVE:
+        PrimitiveTypeInfo pt = (PrimitiveTypeInfo) type;
+        return getter(pt.getBoxed());
+      case BOXED_PRIMITIVE:
+        return "get" + type.getSimpleName();
+      case STRING:
+        return "getString";
+      case JSON_OBJECT:
+        return "getJsonObject";
+      case JSON_ARRAY:
+        return "getJsonArray";
+    }
+    if (type instanceof ClassTypeInfo) {
+      ClassTypeInfo ct = (ClassTypeInfo) type;
+      switch (ct.getName()) {
```

Comment:
MySQL is using `java.time.Duration` type, can we add the mapping here using Row#get(java.time.Duration.class, index)?

Reply:
I will try adding a specific tests for this using test container and implement it.

Reply:
I added support for this and specific integration tests